src/mesh/mesh.cpp

```diff
@@ -1113,18 +1147,23 @@ void Mesh::Initialize(bool init_problem, ParameterInput *pin, ApplicationInput *
     }
 
     //  Now do prolongation, compute primitives, apply BCs
-    for (int i = 0; i < nmb; ++i) {
-      auto &mbd = block_list[i]->meshblock_data.Get();
-      ApplyBoundaryConditions(mbd);
-      // Call MeshBlockData based FillDerived functions
-      Update::FillDerived(mbd.get());
-    }
     for (int i = 0; i < num_partitions; i++) {
       auto &md = mesh_data.GetOrAdd("base", i);
+      if (multilevel) {
+        ApplyCoarseBoundaryConditions(md);
+        ProlongateBoundaries(md);
+      }
+      ApplyFineBoundaryConditions(md);
```

Comment:
From `ApplyFineBoundaryConditions` above:
```c++
if (!md->GetMeshPointer()->multilevel) return TaskStatus::complete;
```

Does this logic check out? 
IIRC we also want the fine boundary conditions to be always set because those are the ones next to the active zones, isn't it?

Reply:
Good catch, that was a left over from copy/paste of the coarse boundary conditions. I fixed this on another branch (after debugging for an embarrassingly long time), but I guess the fix never made it over here.

Reply:
This is fixed by switching to `ApplyBoundaryConditionsOnCoarseOrFineMD`.